tests/Unit/Domain/Test_DomainCreators.cpp

```diff
@@ -0,0 +1,194 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <algorithm>
+#include <catch.hpp>
+
+#include "Domain/Block.hpp"
+#include "Domain/DomainHelpers.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+#include <iostream>
+#include "DataStructures/IndexIterator.hpp"
+
+template <size_t VolumeDim>
+class CornerIterator {
+ public:
+  CornerIterator();
```

Comment:
document the two constructors .  (i.e. the first iterates over all 2^Dim corners; the second over the 2^(Dim-1) corners of the face in the given direction)